client-spark/spark3/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java

```diff
@@ -259,12 +262,13 @@ public void write(Iterator<Product2<K, V>> records) {
       if (shuffleManager.isRssResubmitStage()) {
         throwFetchFailedIfNecessary(e);
       } else {
-        throw e;
+        throw new RssException(e);
       }
     }
   }
 
-  private void writeImpl(Iterator<Product2<K, V>> records) {
+  // gluten need this method and IOException must be throws here
```

Comment:
```suggestion
  // gluten need this method and IOException must be thrown here
```

Reply:
Implementations of this method can catch any `IOException` and rethrow a `new RssException(e)`, this would keep the interface clean.

Reply:
Done

Reply:
> Implementations of this method can catch any `IOException` and rethrow a `new RssException(e)`, this would keep the interface clean.

@xianjingfeng I did not mean to add that as a comment but to apply this to your gluten client implementation. There is no need for this method to specifically throw an `IOException`, simply throw an `RssException` inside your implementation of `writeImpl`.

Reply:
But the current version of gluten can't be supported.

Reply:
Are you saying without `throws IOException`, the current version of gluten can't be supported? Why is this?

Reply:
ok for me

Reply:
@EnricoMi Done. PTAL.